src/gui/MainWindow.cpp

```diff
@@ -190,7 +190,21 @@ MainWindow::MainWindow()
 
 #ifdef WITH_XC_SSHAGENT
     connect(sshAgent(), SIGNAL(error(QString)), this, SLOT(showErrorMessage(QString)));
+    connect(sshAgent(), SIGNAL(enabledChanged(bool)), this, SLOT(agentEnabled(bool)));
     m_ui->settingsWidget->addSettingsPage(new AgentSettingsPage(m_ui->tabWidget));
+
+    m_entryContextMenu->addSeparator();
+    m_entryContextMenu->addAction(m_ui->actionEntryAddToAgent);
+    m_entryContextMenu->addAction(m_ui->actionEntryRemoveFromAgent);
+
+    m_ui->actionEntryAddToAgent->setIcon(filePath()->icon("apps", "utilities-terminal"));
+    m_ui->actionEntryRemoveFromAgent->setIcon(filePath()->icon("apps", "utilities-terminal"));
```

Comment:
This needs to be adjusted once #4411 is merged. I would prefer we halt this until then, since the conflict is easier to fix here.

Reply:
Is done.